firmware/fw_base.S

```diff
@@ -27,6 +27,64 @@ _start:
 
 	li	ra, 0
 	call	_reset_regs
+	/* Preload per-HART details
+	 * s6 -> HART ID
+	 * s7 -> HART Count
+	 * s8 -> HART Stack Size
+	 */
+	csrr	s6, CSR_MHARTID
```

Comment:
The s6 is not used anywhere in the scratch_init loop. Please remove this instruction.